node_modules/oae-principals/lib/internal/dao.js

```diff
@@ -169,10 +213,11 @@ var getPrincipals = module.exports.getPrincipals = function(principalIds, fields
  *
  * @param  {String}     principalId     The id of the principal to update
  * @param  {Object}     profileFields   An object containing the profile field updates to apply
+ * @param  {String}     [oldEmail]      The old email address that was associated with the user. If this is defined and different from the `profileFields.email` address, the email to principal mapping will be updated
```

Comment:
I think at the least, rather than having a new parameter for "old email", we can just fetch the full principal within the `updatePrincipal` method if the `profileFields` object contains an `email` key.